src/iden3comm/handlers/discovery-protocol.ts

```diff
@@ -0,0 +1,198 @@
+import { PROTOCOL_MESSAGE_TYPE } from '../constants';
+
+import { BasicMessage, IPackageManager } from '../types';
+
+import * as uuid from 'uuid';
+import {
+  DiscoverFeatureDiscloseMessage,
+  DiscoverFeatureDisclosure,
+  DiscoverFeatureQueriesMessage,
+  DiscoveryProtocolFeatureType
+} from '../types/protocol/discovery-protocol';
+import {
+  AbstractMessageHandler,
+  BasicHandlerOptions,
+  IProtocolMessageHandler
+} from './message-handler';
+import { getUnixTimestamp } from '@iden3/js-iden3-core';
+import { verifyExpiresTime } from './common';
+
+/**
+ * @beta
+ * DiscoveryProtocolOptions contains options for DiscoveryProtocolHandler
+ * @public
+ * @interface   DiscoveryProtocolOptions
+ */
+export interface DiscoveryProtocolOptions {
+  packageManager: IPackageManager;
+}
+
+/**
+ *
+ * Options to pass to discovery-protocol handler
+ *
+ * @beta
+ * @public
+ * @type DiscoveryProtocolHandlerOptions
+ */
+export type DiscoveryProtocolHandlerOptions = BasicHandlerOptions & {
+  disclosureExpiresDate?: Date;
+};
+
+/**
+ * @beta
+ * createDiscoveryFeatureQueryMessage is a function to create didcomm protocol discovery-feature query message
+ * @param opts - discovery-feature query options
+ * @returns `DiscoverFeatureQueriesMessage`
+ */
+export function createDiscoveryFeatureQueryMessage(opts?: {
+  featureTypes?: (DiscoveryProtocolFeatureType | string)[];
+  from?: string;
```

Comment:
should `from` and `to` be optional?

Reply:
I think yes, it's not mandatory for this message type